ticketbuyer/buyer.go

```diff
@@ -104,6 +101,198 @@ type TicketPurchaser struct {
 	balEstimated   dcrutil.Amount
 	ticketPrice    dcrutil.Amount
 	ProportionLive float64
+
+	// purchaserMtx protects the following runtime configurable options.
+	purchaserMtx      sync.Mutex
+	accountName       string
+	balanceToMaintain dcrutil.Amount
+	maxPriceAbsolute  dcrutil.Amount
+	maxPriceRelative  float64
+	maxFee            dcrutil.Amount
+	poolFees          float64
+	maxPerBlock       int
+}
+
+// Config returns the current ticket buyer configuration.
+func (t *TicketPurchaser) Config() *Config {
+	t.purchaserMtx.Lock()
+	var poolAddress string
+	if t.poolAddress != nil {
+		poolAddress = t.poolAddress.String()
+	}
+
+	config := &Config{
+		AccountName:               t.accountName,
+		AvgPriceMode:              t.cfg.AvgPriceMode,
+		AvgPriceVWAPDelta:         t.cfg.AvgPriceVWAPDelta,
+		BalanceToMaintainAbsolute: t.balanceToMaintain,
+		BlocksToAvg:               t.cfg.BlocksToAvg,
+		DontWaitForTickets:        t.cfg.DontWaitForTickets,
+		ExpiryDelta:               t.cfg.ExpiryDelta,
+		FeeSource:                 t.cfg.FeeSource,
+		FeeTargetScaling:          t.cfg.FeeTargetScaling,
+		MinFee:                    t.cfg.MinFee,
+		MinPriceScale:             t.cfg.MinPriceScale,
+		MaxFee:                    t.maxFee,
+		MaxPerBlock:               t.maxPerBlock,
+		MaxPriceAbsolute:          t.maxPriceAbsolute,
+		MaxPriceRelative:          t.maxPriceRelative,
+		MaxPriceScale:             t.cfg.MaxPriceScale,
+		MaxInMempool:              t.cfg.MaxInMempool,
+		PoolAddress:               poolAddress,
+		PoolFees:                  t.poolFees,
+		SpreadTicketPurchases:     t.cfg.SpreadTicketPurchases,
+		TicketAddress:             t.cfg.TicketAddress,
+		TxFee:                     t.cfg.TxFee,
+	}
+	t.purchaserMtx.Unlock()
+	return config
+}
+
+// AccountName returns the account name to use for purchasing tickets.
+func (t *TicketPurchaser) AccountName() string {
+	t.purchaserMtx.Lock()
+	accountName := t.accountName
+	t.purchaserMtx.Unlock()
+	return accountName
+}
+
+// SetAccountName sets the account name to use for purchasing tickets.
+func (t *TicketPurchaser) SetAccountName(accountName string) {
```

Comment:
this needs some sort of validation that the account exists.

We should also be using account numbers to refer to accounts, never names.  Do the conversion from names to numbers in the rpc server handlers.